### PR TITLE
Add more sigchain data verification

### PR DESCRIPTION
**Pull request by yilunzhang**
Add more sigchain data verification

### Type (put an `x` where ever applicable)
- [ ] Bug fix: Link to the issue
- [x] Feature (Non-breaking change)
- [ ] Feature (Breaking change)
- [ ] Documentation Improvement

### CODE REVIEW

File: pb/sigchain.go [hunk 1]
```diff
@@ -141,6 +145,56 @@ func (sc *SigChain) Verify() error {
...
+
+	if !sc.IsComplete() {
+		return fmt.Errorf("sigchain is not complete")
+	}
+
+	if bytes.Equal(sc.Elems[0].Id, sc.Elems[1].Id) {
+		return fmt.Errorf("sender and relayer 1 has the same ID")
+	}
+
+	if sc.IsComplete() && len(sc.Elems[sc.Length()-1].NextPubkey) > 0 {
```
Comment (gdmmx) [inline pb/sigchain.go]: Why sc.IsComplete() had double checked?

File: pb/sigchain.go [hunk 2]
```diff
@@ -141,6 +145,56 @@ func (sc *SigChain) Verify() error {
...
+	if sc.Elems[0].Mining {
+		return fmt.Errorf("first sigchain element should have mining set to false")
+	}
+
+	if sc.Elems[sc.Length()-1].Mining {
+		return fmt.Errorf("last sigchain element should have mining set to false")
+	}
+
+	for i, e := range sc.Elems {
+		if i == 0 || (sc.IsComplete() && i == sc.Length()-1) {
```
Comment (gdmmx) [inline pb/sigchain.go]: The third time `sc.IsComplete()` checked.

**Comment by NoelBright** [comment]
looks good to me

File: pb/sigchain.go (hunk 1 above)
Comment (yilunzhang) [reply pb/sigchain.go]: Removed duplicate check

File: pb/sigchain.go (hunk 2 above)
Comment (yilunzhang) [reply pb/sigchain.go]: Removed duplicate check

**Pull request merged**